R/meta_regression/regression_plot_panel.R

```diff
@@ -0,0 +1,276 @@
+.CreateInlineBlock <- function(..., style = NULL) {
+  return(
+    div(
+      ...,
+      style = paste0("display: inline-block; ", style)
+    )
+  )
+}
+
+.AddTooltip <- function(..., tooltip) {
+  return(
+    div(
+      ...,
+      title = tooltip
+    )
+  )
+}
+
+#' Create the regression plot panel.
+#'
+#' @param id ID of the module.
+#' @return Div containing the module UI.
+regression_plot_panel_ui <- function(id) {
+  ns <- NS(id)
+  
+  sidebarLayout(
+    position = "right",
+    mainPanel = mainPanel(
+      width = 9,
+      plotOutput(outputId = ns("regression_plot"), height = "800px")
+    ),
+    sidebarPanel = sidebarPanel(
+      width = 3,
+      # Add comparators
+      selectInput(inputId = ns("add_comparator_dropdown"), label = "Add Comparator", choices = c(), selectize = FALSE),
+      actionButton(inputId = ns("add_comparator_btn"), label = "Add to plot"),
+      div(
+        style = "float: right;",
+        actionButton(inputId = ns("add_all_btn"), label = "Add all")
+      ),
+      br(),
+      br(),
+      # Remove comparators
+      selectInput(inputId = ns("remove_comparator_dropdown"), label = "Remove Comparator", choices = c(), selectize = FALSE),
+      actionButton(inputId = ns("remove_comparator_btn"), label = "Remove"),
+      div(
+        style = "float: right;",
+        actionButton(inputId = ns("remove_all_btn"), label = "Remove all")
+      ),
+      
+      h3("Plot Options"),
+      # Covariate value
+      .AddTooltip(
+        checkboxInput(inputId = ns("covariate"), label = "Show covariate value", value = TRUE),
+        tooltip = "Show the covariate value as a vertical line at the current value"
+      ),
+      # Confidence regions
+      .AddTooltip(
+        checkboxInput(inputId = ns("confidence"), label = "Show confidence regions"),
+        tooltip = "Show confidence regions for the added comparisons"
+      ),
+      sliderInput(
+        inputId = ns("confidence_opacity"),
+        label = "Confidence Region Opacity",
+        min = 0,
+        max = 0.5,
+        step = 0.01,
+        value = 0.2
+      ),
+      # Regression lines
+      .AddTooltip(
+        checkboxInput(inputId = ns("ghosts"), label = "Show ghost comparisons", value = TRUE),
+        tooltip = "Show other available comparisons in light grey behind the main plot"
+      ),
+      .AddTooltip(
+        checkboxInput(inputId = ns("extrapolate"), label = "Extrapolate regression lines", value = TRUE),
+        tooltip = "Extrapolate the regression lines beyond the range of the original data"
+      ),
+      # Contributions
+      .AddTooltip(
+        checkboxInput(inputId = ns("contributions"), label = "Show contributions", value = TRUE),
+        tooltip = "Show study contributions as circles, where a bigger circle represents a larger contribution"
+      ),
+      
+      div(
+        id = ns("contribution_options"),
+        p("Study circle sized by:"),
```

Comment:
This doesn't get "ghosted out" like the slider which makes it seem like it's in use all the time

Reply:
That's a limitation of the widget. I've changed it to be a standard input, so it behaves now.